libgit/autogit_manager.go

```diff
@@ -191,23 +217,54 @@ func (am *AutogitManager) LocalChange(
 }
 
 func (am *AutogitManager) getNodesToInvalidate(
-	affectedNodeIDs []libkbfs.NodeID) (nodes []libkbfs.Node) {
+	affectedNodeIDs []libkbfs.NodeID) (
+	nodes []libkbfs.Node, repoNodeIDs []libkbfs.NodeID) {
 	am.registryLock.RLock()
 	defer am.registryLock.RUnlock()
 	for _, nodeID := range affectedNodeIDs {
 		node, ok := am.repoNodesForWatchedIDs[nodeID]
 		if ok {
 			nodes = append(nodes, node)
+			repoNodeIDs = append(repoNodeIDs, nodeID)
+		}
+	}
+	return nodes, repoNodeIDs
+}
+
+func (am *AutogitManager) clearInvalidatedBrowsers(
+	repoNodeIDs []libkbfs.NodeID) {
```

Comment:
Would it be better to make `repoNodeIDs` a `map` to make the `rootNodeID` check below faster?

Reply:
Ah I see. I was somehow under the impression that it may contain thousands of nodes. If it's very small then this is good. Thanks for explaining! (Though it might be helpful to add a comment when you touch this file again.)